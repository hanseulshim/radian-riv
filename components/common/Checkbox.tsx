import React from 'react'
interface Props {
  label: string
  checked: boolean
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  disabled?: boolean
}

export default function Checkbox({
  label,
  checked,
  onChange,
  disabled,
  ...props
}: Props) {
  return (
    <div className="checkbox-container">
      <label>
        <input
          type="checkbox"
          className="hidden-checkbox"
          checked={checked}
          onChange={onChange}
          disabled={disabled}
          {...props}
        />
        <div
          className={`styled-checkbox ${checked && 'checked'} ${
            disabled && 'disabled'
          }`}
        >
          <svg viewBox="0 0 24 24">
            <polyline
              className={`checkmark ${!checked && 'hidden'}`}
              points="20 6 9 17 4 12"
            />
          </svg>
        </div>
        <span className={`${disabled && 'disabled'}`}>{label}</span>
      </label>
    </div>
  )
}
