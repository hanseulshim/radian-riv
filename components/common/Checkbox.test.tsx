import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Checkbox from './Checkbox'

describe('Checkbox', () => {
  let component
  beforeAll(() => {
    const props = {
      label: 'Test',
      checked: false,
      onChange: () => {},
      disabled: true
    }
    component = shallow(<Checkbox {...props} />)
  })
  it('has a div with checked', () => {
    expect(component.find('.styled-checkbox')).toHaveLength(1)
    expect(component.find('.checked')).toHaveLength(0)
  })
  it('has hidden class', () => {
    expect(component.find('.hidden')).toHaveLength(1)
  })
  it('label should be disabled', () => {
    expect(component.find('span.disabled')).toHaveLength(1)
  })
  it('label should match value passed in', () => {
    expect(component.find('span.disabled').text()).toEqual('Test')
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
