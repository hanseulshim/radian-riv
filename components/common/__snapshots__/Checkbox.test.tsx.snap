// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match snapshot 1`] = `
<div
  className="checkbox-container"
>
  <label>
    <input
      checked={false}
      className="hidden-checkbox"
      disabled={true}
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="styled-checkbox disabled"
    >
      <svg
        viewBox="0 0 24 24"
      >
        <polyline
          className="checkmark hidden"
          points="20 6 9 17 4 12"
        />
      </svg>
    </div>
    <span
      className="label disabled"
    >
      Test
    </span>
  </label>
</div>
`;
